external-crates/move/move-binary-format/src/compatibility.rs

```diff
@@ -209,9 +215,17 @@ impl Compatibility {
 }
 
 // When upgrading, the new abilities must be a superset of the old abilities.
-// Adding an ability is fine, but removing an ability could cause existing usages to fail.
-fn struct_abilities_compatibile(old_abilities: AbilitySet, new_abilities: AbilitySet) -> bool {
+// Adding an ability is fine as long as it's not in the disallowed_new_abilities,
+// but removing an ability could cause existing usages to fail.
+fn struct_abilities_compatibile(
+    disallowed_new_abilities: AbilitySet,
+    old_abilities: AbilitySet,
+    new_abilities: AbilitySet,
+) -> bool {
     old_abilities.is_subset(new_abilities)
+        && disallowed_new_abilities.into_iter().all(|ability| {
+            !new_abilities.has_ability(ability) || old_abilities.has_ability(ability)
```

Comment:
extremely minor nit, it just bothers me that languages don't have implication and I always have to work it out in my head, can you just comment `new_abilities.has_ability(ability) ==> old_abilities.has_ability(ability)`
Or in english
`// If new abilities has the ability, then the old abilities must have it too`